### ISSUE TITLE
Number of front-to-backed drawables can be increased

### ISSUE THREAD

**Issue by smoogipoo** [ppy/osu-framework#3751]
Currently, the FTB pass only supports 2000 drawables https://github.com/ppy/osu-framework/blob/63f953d3844d28acfc34354fef8e769dd5d648b4/osu.Framework/Graphics/OpenGL/DepthValue.cs#L23

The implementation there is very approximate so that it behaves correctly on all hardware/all depth buffers, however it should be possible to support an arbitrary bit depth with FP bit increments. I've reconstructed `DepthValue` with 16b increments here: https://gist.github.com/smoogipoo/76e2739a35f88ce790e495504c2db9b1

The above example works up until ~17000 boxes in `TestSceneFrontToBack`, at which point it starts behaving weirdly.

**Comment by peppy** [comment]
You mentioned elsewhere that there was a remaining issue with the change, is that the "weird" behaviour you noticed?

**Comment by smoogipoo** [comment]
Yep, that's what I'm referring to. It's z-fighting at values very close to 0, possibly because I was using fp16 values in an fp24 depth buffer.

**Comment by peppy** [comment]
I see. Wonder if we can just cut a bit short and avoid that but still bump up the limit.
